src/Datadog.Trace/SpanContext.cs

```diff
@@ -100,6 +116,11 @@ private SpanContext(TraceId? traceId, string serviceName)
         /// </summary>
         internal string Origin { get; }
 
+        /// <summary>
+        /// Gets the trace state for W3C propagation
+        /// </summary>
+        internal string TraceState { get; }
```

Comment:
this is not following the OTEL model: https://github.com/open-telemetry/opentelemetry-specification/blob/cf59c0b868744b764ca8b9175668abc6755fecca/specification/trace/api.md#spancontext
See an example implementation here: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/Trace/SpanContext.cs

Reply:
I have redone how this is parsed.